src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SslOverTdsStream.cs

```diff
@@ -160,7 +161,12 @@ public override void SetLength(long value)
         /// </summary>
         public override void Flush()
         {
-            _stream.Flush();
+            // Can sometimes get Pipe broken errors from flushing a PipeStream.
+            // PipeStream.Flush() also doesn't do anything, anyway.
+            if (!(_stream is PipeStream))
```

Comment:
If this is the only type it'll use, and if the broken pipe error happens frequently, yes.